src/perfmon.cpp

```diff
@@ -0,0 +1,614 @@
+#include "perfmon.h"
+
+#include <map>
+#include <string>
+#include <sstream>
+#include <iomanip>
+#include <vector>
+#include <ctime>
+#include <cfloat>
+#include <cstring>
+
+#include <sys/time.h>
+
+static double const INFINITY = DBL_MAX;
+
+#define PULSE_PER_SECOND 10
```

Comment:
Should this reference awake.h's PASSES_PER_SEC so it scales if someone changes the pulse frequency later?

Reply:
My design preference has been to have it as a standalone module with no direct dependencies on the game code. Could make this configurable from outside instead of a define if you think that's worthwhile.

Reply:
This is fine for now, I'll revisit later if we consider changing the pass value.